modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -982,6 +987,43 @@ else if (!spi.failureDetectionTimeoutEnabled() && reconCnt == spi.getReconnectCo
         }
     }
 
+    /**
+     * Pings the node by its address to see if it's alive.
+     *
+     * @param addr Address of the node.
+     * @param nodeId Node ID to ping. In case when client node ID is not null this node ID is an ID of the router node.
+     * @param clientNodeId Client node ID.
+     * @param timeout Timeout on operation in milliseconds. If 0, a value based on {@link TcpDiscoverySpi} is used.
+     * @param debugLogging Boolean flag indicating whether debug information should be printed into node log.
+     * @return ID of the remote node and "client exists" flag if node alive or {@code null} if the remote node has
+     *         left a topology during the ping process.
+     * @throws IgniteCheckedException If an error occurs.
+     * @see #pingNodeImpl(InetSocketAddress, UUID, UUID, long, boolean)
+     */
+    @Nullable private IgniteBiTuple<UUID, Boolean> pingNode(
+            InetSocketAddress addr,
+            @Nullable UUID nodeId,
+            @Nullable UUID clientNodeId,
+            long timeout,
+            boolean debugLogging) throws IgniteCheckedException {
+        try {
+            return pingNodeImpl(addr, nodeId, clientNodeId, timeout, debugLogging);
+        }
+        catch (IgniteCheckedException e) {
+            logPingError(e.getMessage(), debugLogging);
+
+            throw e;
+        }
+    }
```

Comment:
Additional wrapper is redundant. Let's rename `pingNodeImpl` back to `pingNode` and use it with one additional parameter

Reply:
thank you, I agree